std/algorithm/sorting.d

```diff
@@ -2166,6 +2167,131 @@ unittest
     assert(isSorted!(binaryFun!("a < b"))(b));
 }
 
+/**
+Given a range of elements, constructs an index of its top $(I n) elements
+(i.e., the first $(I n) elements if the range were sorted).
+
+Similar to $(LREF topN), except that the range is not modified.
+
+Params:
+    less = A binary predicate that defines the ordering of range elements.
+        Defaults to $(D a < b).
+    ss = $(RED (Not implemented yet.)) Specify the swapping strategy.
+    r = A $(XREF2 range, isRandomAccessRange, random-access range) of elements
+        to make an index for.
+    index = A $(XREF2 range, isRandomAccessRange, random-access range) with
+        assignable elements to build the index in. The length of this range
+        determines how many top elements to index in $(D r).
+
+        This index range can either have integral elements, in which case the
+        constructed index will consist of zero-based numerical indices into
+        $(D r); or it can have pointers to the element type of $(D r), in which
+        case the constructed index will be pointers to the top elements in
+        $(D r).
+    sorted = Determines whether to sort the index by the elements they refer
+        to.
+
+See_also: $(LREF topN), $(LREF topNCopy).
+
+BUGS:
+The swapping strategy parameter is not implemented yet; currently it is
+ignored.
```

Comment:
Better: enforce `unstable` in a static assert at the top of the implementation. Then the algorithm is correct but incomplete.